app/src/main/java/org/p2p/wallet/common/ui/widget/EarnWidget.kt

```diff
@@ -0,0 +1,183 @@
+
+    init {
+        binding.tickerViewAmount.apply {
+            setCharacterLists(TickerUtils.provideNumberList())
+            setPreferredScrollingDirection(TickerView.ScrollingDirection.DOWN)
+        }
+    }
+
+    fun setOnButtonClickListener(buttonClickListener: OnClickListener) {
+        binding.buttonEarn.setOnClickListener(buttonClickListener)
+    }
+
+    fun setWidgetState(state: EarnWidgetState) = with(binding) {
+        currentState = state
+        viewEarnContent.setBackgroundColor(getColor(state.backgroundColor))
+        val balanceState = state is EarnWidgetState.Balance
+        val depositingState = state is EarnWidgetState.Depositing
+        textViewEarnTitle.isVisible = !depositingState
+        tickerViewAmount.isVisible = balanceState
+        textViewEarnMessage.isVisible = !balanceState
+        viewTokenContainer.isVisible = balanceState
+        buttonEarn.isEnabled = !depositingState
+        when (state) {
+            is EarnWidgetState.Balance -> {
+                makeAlignStartContent()
+                tickerViewAmount.text = "$${state.amount.formatTicker()}"
+                textViewEarnTitle.setText(R.string.earn_widget_balance_title)
+                buttonEarn.setText(R.string.earn_widget_balance_button)
+            }
+            is EarnWidgetState.LearnMore -> {
+                makeAlignCenterContent()
+                textViewEarnTitle.setText(R.string.earn_widget_learn_more_title)
+                textViewEarnMessage.setText(R.string.earn_widget_learn_more_message)
+                buttonEarn.setText(R.string.earn_widget_learn_more_button)
+            }
+            is EarnWidgetState.Depositing -> {
+                makeAlignStartContent()
+                tickerViewAmount.isVisible = false
+                textViewEarnTitle.isVisible = false
+                textViewEarnMessage.isVisible = false
+                buttonEarn.setText(state.buttonTextRes)
+                buttonEarn.isEnabled = true // TODO remove after tests and real integration
+            }
+            is EarnWidgetState.DepositFoundsFailed -> {
+                makeAlignCenterContent()
+                textViewEarnTitle.setText(R.string.earn_widget_deposit_failed_title)
+                textViewEarnMessage.setText(R.string.earn_widget_deposit_failed_message)
+                buttonEarn.setText(R.string.earn_widget_deposit_failed_button)
+            }
+            is EarnWidgetState.Error -> {
+                makeAlignCenterContent()
+                textViewEarnTitle.setText(R.string.earn_widget_error_title)
+                textViewEarnMessage.setText(state.messageTextRes)
+                buttonEarn.setText(state.buttonTextRes)
+            }
+        }
+    }
+
+    private fun makeAlignStartContent() = with(binding) {
+        textViewEarnTitle.apply {
+            textAlignment = TEXT_ALIGNMENT_TEXT_START
+            setTextAppearance(R.style.UiKit_TextAppearance_Regular_Text3)
```

Comment:
Why here?
What is the problem setting this in xml?

Reply:
because title has 2 different styles according to our designs

Reply:
=(